apps/rush/src/taskRunner/ProjectBuildTask.ts

```diff
@@ -220,34 +213,38 @@ export default class ProjectBuildTask implements ITaskDefinition {
     }
   }
 
-  private _getBuildCommand(): IBuildCommand {
-    const build: IBuildCommand =
+  private _getBuildCommand(): string {
+    const build: string =
       this._getScriptCommand('test') || this._getScriptCommand('build');
 
-    if (!build) {
+    if (build === undefined) {
       // tslint:disable-next-line:max-line-length
       throw new Error(`The project [${this._rushProject.packageName}] does not define a 'test' or 'build' command in the 'scripts' section of its package.json`);
     }
 
-    if (build.args) {
-      // Normalize test command step
-      build.args.push(this._errorDisplayMode === ErrorDetectionMode.VisualStudioOnline ? '--no-color' : '--color');
+    if (build === '') {
+      return build;
+    }
 
-      if (this._production) {
-        build.args.push('--production');
-      }
-      if (this._npmMode) {
-        build.args.push('--npm');
-      }
-      if (this._minimalMode) {
-        build.args.push('--minimal');
-      }
+    // Normalize test command step
+    const args: string[] = [];
+
+    args.push(this._errorDisplayMode === ErrorDetectionMode.VisualStudioOnline ? '--no-color' : '--color');
+
+    if (this._production) {
+      args.push('--production');
+    }
+    if (this._npmMode) {
+      args.push('--npm');
+    }
+    if (this._minimalMode) {
+      args.push('--minimal');
     }
 
-    return build;
+    return `${build} ${args.join(' ')}`;
```

Comment:
>${args.join(' ')} [](start = 21, length = 17)

This is not escaped properly.  In order to escape it, you need to know which shell you're talking to.

I guess it's a hack, because our long term plan was to use a different mechanism (e.g. environment variables).